### PR TITLE
Add clcache for windows

### PR DESCRIPTION
**Pull request by OwenArnold**
Need see upstream pipelines https://github.com/scipp/pipelines/pull/8

| Build Type  | Cmake build time in Minutes |
| ------------- | ------------- |
| No clcache  | 27 |
| Cold clcache (All miss) |  41 |
| Hot clcache  (No miss) |  19  |

### CODE REVIEW

**Comment by SimonHeybrock** [comment]
Why does a cold cache increase build times so significantly?

**Comment by OwenArnold** [comment]
> Why does a cold cache increase build times so significantly?

I assume it's the IO associated with the caching process. clcache does have a profile option that may reveal this kind of thing, but I have not run it with the profiler enabled.

File: .azure-pipelines/pull_request.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ resources:
     name: scipp/pipelines
     type: github
     endpoint: scipp
-    ref: main
+    ref: clcache # TODO change back to main after piplines branch merged in that repo
```
Comment (SimonHeybrock) [inline .azure-pipelines/pull_request.yml:17]:
```suggestion
    ref: main
```

**Comment by SimonHeybrock** [comment]
Should merge this ASAP, because the upstream merge in `pipelines` appears to break `main`: https://dev.azure.com/scipp/scipp/_build/results?buildId=6823&view=logs&j=d5c58a2e-1ae4-501e-e421-b21c980eb30a&t=fcbe8920-cf97-5746-799c-eac6f4530465.

**Pull request merged**